### PR TITLE
Move to intra-doc links for task.rs and vec.rs

### PR DESCRIPTION
**Pull request by kofls**
Partial fix for #75080 

links for [`get`], [`get_mut`] skipped due to #75643 
link for [`copy_from_slice`] skipped due to #63351

### CODE REVIEW

File: library/alloc/src/vec.rs [hunk 1]
```diff
@@ -278,22 +275,18 @@ use crate::raw_vec::RawVec;
 /// `Vec` does not currently guarantee the order in which elements are dropped.
 /// The order has changed in the past and may change again.
 ///
-/// [`vec!`]: ../../std/macro.vec.html
 /// [`get`]: ../../std/vec/struct.Vec.html#method.get
 /// [`get_mut`]: ../../std/vec/struct.Vec.html#method.get_mut
```
Comment (kofls) [inline library/alloc/src/vec.rs:279]: skipped due to #75643

File: library/alloc/src/vec.rs [hunk 2]
```diff
@@ -867,7 +860,7 @@ impl<T> Vec<T> {
     ///
     /// [`truncate`]: #method.truncate
     /// [`resize`]: #method.resize
```
Comment (jyn514) [inline library/alloc/src/vec.rs:862]: These are implemented on `Vec`, not on slices.
```suggestion
    /// [`truncate`]: Vec::truncate
    /// [`resize`]: Vec::resize
```

File: library/alloc/src/vec.rs [hunk 3]
```diff
@@ -867,7 +860,7 @@ impl<T> Vec<T> {
...
-    /// [`extend`]: ../../std/iter/trait.Extend.html#tymethod.extend
+    /// [`extend`]: Extend::extend
     /// [`clear`]: #method.clear
```
Comment (jyn514) [inline library/alloc/src/vec.rs:864]:
```suggestion
    /// [`clear`]: Vec::clear
```

File: library/alloc/src/vec.rs [hunk 4]
```diff
@@ -1480,7 +1471,6 @@ impl<T> Vec<T> {
     /// ```
     ///
     /// [`resize`]: #method.resize
```
Comment (jyn514) [inline library/alloc/src/vec.rs:1473]:
```suggestion
    /// [`resize`]: Vec::resize
```

File: library/alloc/src/vec.rs [hunk 5]
```diff
@@ -1590,8 +1580,6 @@ impl<T: Clone> Vec<T> {
     /// assert_eq!(vec, [1, 2]);
     /// ```
     ///
-    /// [`Clone`]: ../../std/clone/trait.Clone.html
-    /// [`Default`]: ../../std/default/trait.Default.html
     /// [`resize_with`]: #method.resize_with
```
Comment (jyn514) [inline library/alloc/src/vec.rs:1583]:
```suggestion
    /// [`resize_with`]: Vec::resize_with
```

File: library/alloc/src/vec.rs [hunk 6]
```diff
@@ -1655,9 +1643,7 @@ impl<T: Default> Vec<T> {
     /// ```
     ///
     /// [`resize`]: #method.resize
-    /// [`Default::default()`]: ../../std/default/trait.Default.html#tymethod.default
-    /// [`Default`]: ../../std/default/trait.Default.html
-    /// [`Clone`]: ../../std/clone/trait.Clone.html
+    /// [`Default::default()`]: Default::default
```
Comment (jyn514) [inline library/alloc/src/vec.rs:1646]: The parentheses count as a disambiguator, they'll still link :)
```suggestion
```

File: library/alloc/src/vec.rs [hunk 7]
```diff
@@ -1655,9 +1643,7 @@ impl<T: Default> Vec<T> {
     /// ```
     ///
     /// [`resize`]: #method.resize
```
Comment (jyn514) [inline library/alloc/src/vec.rs:1645]:
```suggestion
    /// [`resize`]: Vec::resize
```

**Pull request merged**